### ISSUE TITLE
RegExpIndicesArray - Named group indices can be undefined

### ISSUE THREAD

**Issue by TomStrepsil** [microsoft/TypeScript#63281]
### 🔎 Search Terms

RegExpIndicesArray, Regex indices

### 🕗 Version & Regression Information

- This is the behaviour in every version I tried, and I reviewed the FAQ for entries about Regular Expressions

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=9&ts=6.0.0-dev.20260322#code/PTAEGcCcGNgWwIYEsB2A6ALuAUN6B7FcDUAc0nwFcAHUAXlGAQAoB+AHjiQBMA+AIwCUrWNzQBTAB7jozAOQJocwQEI0qbkmjjwa8lWq6A2gCIu3EwF0A3LgJESGqfTIUaRgAw3sQA

### 💻 Code

```ts
const group = /a(?<mid>b)?c/d.exec('ac')!.indices!.groups!["mid"];
const index = group[0];
```
The typing assumes that group indices are of type `[number, number]`, but these can also be `undefined`

### 🙁 Actual behavior

`typeof group === "undefined"`

hence, the type system should have complained when attempting to access via index.

### 🙂 Expected behavior

`'group' is possibly 'undefined'`

...should be reported by the type system.

### Additional information about the issue

https://github.com/microsoft/TypeScript/pull/61079 fixed the same issue for the outer `RegExpIndicesArray` itself, but not the indices array on the matched groups

**Comment by RyanCavanaugh** [comment]
I think this falls under #32098

**Comment by TomStrepsil** [comment]
> I think this falls under [#32098](https://github.com/microsoft/TypeScript/issues/32098)

It'd be great if this could be fixed as part of that, but that feels like a much bigger change - i.e. type safety around the keys of named capture groups themselves.

The example above doesn't really change the reality of keyed groups, the example:
```js
/a(?<mid>b)?c/d.exec('ac').indices.groups;
```
returns:
```js
{ mid: undefined }
```

Having strongly-typed keys seems unrelated to the fact the _value_ is `undefined` rather than `[number, number]`, if the group failed to match.

**Comment by TomStrepsil** [comment]
@RyanCavanaugh - since [the Contribution guide](https://github.com/microsoft/TypeScript?tab=readme-ov-file#contribute) indicates:
> NOTE: Code changes in this repo are now limited to a small category of fixes:
>
> - Crashes that were introduced in 5.9 or 6.0 that also repro in 7.0 and have a portable fix and don't incur other behavioral changes
> - Security issues
> - Language service crashes that substantially impact mainline usage
> - Serious regressions from 5.9 (these must seriously impact a large proportion of users)

...It doesn't feel like this kind of small fix would qualify.

Would you recommend me re-raising this issue at https://github.com/microsoft/typescript-go/ ?

Feels like adding `| undefined` [here](https://github.com/microsoft/typescript-go/blob/281b3d0720ab3de6d330d211705cb0383b47500b/internal/bundled/libs/lib.es2022.regexp.d.ts#L27) would be a quick win.

**Comment by RyanCavanaugh** [comment]
PR rules are about code changes, issues are about behavior changes. This isn't an approved behavior change yet (it'd need to be justified in terms of the breaks it would cause) so PR rules are not relevant.
